xmdsshr/XmdsExpr.c

```diff
@@ -372,7 +372,7 @@ EXPORT struct descriptor *XmdsExprGetXd(Widget w)
       TreeGetDefaultNid(&old_def);
       TreeSetDefaultNid(def_nid);
     }
-    status = (*ew->expr.compile)(&text_dsc, ans MDS_END_ARG);
+    status = TdiCompile((mdsdsc_t *)&text_dsc, ans MDS_END_ARG);
```

Comment:
I would rather solve this with a cast, rather than avoiding using the function pointers at all, this could have larger implications we can't see.